flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServices.java

```diff
@@ -308,7 +309,10 @@ public static TaskManagerServices fromConfiguration(
                         taskManagerServicesConfiguration.getTaskExecutorResourceSpec(),
                         taskManagerServicesConfiguration.getTimerServiceShutdownTimeout(),
                         taskManagerServicesConfiguration.getPageSize(),
-                        ioExecutor);
+                        ioExecutor,
+                        taskManagerServicesConfiguration
+                                .getConfiguration()
+                                .getString(ClusterOptions.TASK_MANAGER_PROCESS_WORKING_DIR_BASE));
```

Comment:
We should use `ClusterOptionsInternal.WORKING_DIRECTORY`.

Reply:
Or even better to use `ClusterEntrypointUtils.getWorkingDirectory(configuration)`.